ambry-network/src/test/java/com.github.ambry.network/NetworkClientTest.java

```diff
@@ -108,6 +124,12 @@ public NetworkClientTest() throws IOException {
     replicaOnSslNode = sslEnabledClusterMap.getReplicaIds(dataNodeId).get(0);
   }
 
+  @After
+  public void cleanup() {
+    // force JVM to gc to trigger more intense resource leak detection.
+    System.gc();
```

Comment:
If there was a leak, would the netty leak detector print messages in the log?

Reply:
Netty resource leak detector works in a weird way. First, it only prints out the error message in the log, without any exception. So it's not considered as an error, thus will not fail the test. It's a bit troubling because if the unit test doesn't fail, we probably not going to read the log even if there is a bytebuf leak. Second, It heavily relies on the garbage collection. resource leak detector only sees a ByteBuf leaking when this ByteBuf is GC-ed, but not released yet. So it's not so predictable.